core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -320,40 +322,86 @@ struct ALL_t {
 namespace Kokkos {
 namespace Impl {
 
+// TODO: Has tuple_size and tuple_size == 2
+// Using inheritance instead of type aliases gives better compile-time error
+// messages in clang.  (Elsewhere it makes no difference).
+template <class T>
+struct is_pairlike_slice
+    : std::integral_constant<bool, has_kokkos_get<T, 0>::value &&
+                                       has_kokkos_get<T, 1>::value> {};
```

Comment:
Doesn't this come back as true for `array<int,3>` ?

Reply:
OK we can't fix this because std::get<2>(std::pair) compiles, and std::tuple_size is not SFINAE friendly.